llvm/lib/Target/AMDGPU/SILowerSGPRSpills.cpp

```diff
@@ -32,12 +33,18 @@ using MBBVector = SmallVector<MachineBasicBlock *, 4>;
 
 namespace {
 
+static cl::opt<unsigned> MaxNumVGPRsForWwmAllocation(
+    "amdgpu-num-vgprs-for-wwm-alloc",
+    cl::desc("Max num VGPRs for whole-wave register allocation."),
+    cl::ReallyHidden, cl::init(5));
```

Comment:
This seems really small and odd for a maximum?

Reply:
I see that. I initially wanted this value between 5 - 7 based on the experience with SGPR spilling on heavy workloads earlier. In the internal testing, I used 5 and 10 for the upper cap and they both worked without any issues. I should have kept the higher value here. Will change it to 10.